makefile

```diff
@@ -741,6 +744,20 @@ style:
 	   "No use of std::cerr found", "Use mfem::err instead of std::cerr");\
 	exit $$err_code
 
+# Generate a TAGS table in $MFEM_DIR from all the tracked files
+tags:
+ifndef ETAGS_BIN
+	$(error Error could not find suitable 'etags', please install one \
+	using your package manager)
+else ifndef EGREP_BIN
+	$(error Error could not find suitable 'egrep', please install one \
+	using your package manager)
+endif
+	$(eval MFEM_TRACKED_SOURCE = $(shell git -C $(MFEM_REAL_DIR) \
+	ls-files | $(EGREP_BIN) -v \(^\(data/\)\|\.\(png\|pdf\|ps\|ppt\|jpg\|txt\|mesh\|yml\|md\)\$\)))
```

Comment:
With the exception of `.png`,`.txt`, and `.mesh`, there really shouldn't be any files with these extensions in the MFEM repo...

Instead of excluding files, isn't is simpler to just include the `*.hpp`, `*.cpp` and `*.h` sources?

Reply:
> instead of excluding files, isn't is simpler to just include the *.hpp, *.cpp and *.h sources?

Are those the only kind of `src` extensions that mfem uses (or will ever use)? Benefit of exclusion over inclusion is that the `TAGS` will work on all sorts of files (e.g. `makefiles`) out of the box without devs needing to remember to add regex to some obscure make target.

Reply:
The MFEM sources are C++ files with `*.hpp` and `*.cpp` extensions. 

There are also a few `*.h` files which are either external tools: `general/tinyxml2.h` and `fem/picojson.h` or C files needed by libCEED: `fem/ceed/*.h`.

Currently I don't see any reason to extend the source extension beyond these, and this assumption is actually used already in the makefile -- see the `style` target and the `FORMAT_FILES` variable.